v2/pkg/genruntime/kubernetes_resource.go

```diff
@@ -37,6 +38,9 @@ type KubernetesResource interface {
 	// 	SetAzureName(name string)
 	// They do not if the name must be a fixed value (like 'default').
 
+	//GroupVersionKind returns a GroupVersionKind of the resource
+	GroupVersionKind() schema.GroupVersionKind
```

Comment:
Where is the actual implementation of this method?

I _think_ this isn't going to do what you want because of the way that k8s populates the GVK. Basically it strips it from the object when it deserializes it, and it's also empty for empty objects, so you'll get an empty struct result. You should be able to test it in a debugger and check.

The way to get the GVK from a given `client.Object` is using the `scheme`, like this: `gvk, err := apiutil.GVKForObject(obj, scheme)`.

I _think_ that's what you'll end up having to do. This is all basically because the GVK is just a 3-tuple to identify the type, so they assume that if you already have the type you basically don't need it. In our case we do for things like this so we have to jump through a few hoops to actually get it.

Reply:
have removed the above! Thought it might work with getting the GVK, however, yes you're right - it doesn't have any implementation hence gonna be empty always.